proc-macros/src/render_server.rs

```diff
@@ -383,18 +384,30 @@ impl RpcDescription {
 			let serde = self.jrps_server_item(quote! { core::__reexports::serde });
 			let serde_crate = serde.to_string();
 
-			let fields = params.iter().zip(generics.clone()).map(|((name, _), ty)| {
+			let fields = params.iter().zip(generics.clone()).map(|(fn_arg, ty)| {
+				let arg_pat = fn_arg.arg_pat();
+				let rename_to = fn_arg.rename_to();
+
 				let mut alias_vals = String::new();
 				alias_vals.push_str(&format!(
 					r#"alias = "{}""#,
-					heck::ToSnakeCase::to_snake_case(name.ident.to_string().as_str())
+					heck::ToSnakeCase::to_snake_case(
+						rename_to.as_deref().unwrap_or(arg_pat.ident.to_string().as_str())
+					)
 				));
 				alias_vals.push(',');
 				alias_vals.push_str(&format!(
 					r#"alias = "{}""#,
-					heck::ToLowerCamelCase::to_lower_camel_case(name.ident.to_string().as_str())
+					heck::ToLowerCamelCase::to_lower_camel_case(
+						rename_to.as_deref().unwrap_or(arg_pat.ident.to_string().as_str())
+					)
 				));
 
+				let serde_rename = rename_to
```

Comment:
we shouldn't need this serde rename would be better to rename the name in the struct instead, if you agree with my proposed API with a function name we could do this instead:

```rust
			let fields = params.iter().zip(generics.clone()).map(|(fn_arg, ty)| {
				// this could be a renamed name or not... but shouldn't matter at this point
                let name = fn_arg.name();

				let mut alias_vals = String::new();
				alias_vals.push_str(&format!(r#"alias = "{}""#, heck::ToSnakeCase::to_snake_case(name.as_str())));
				alias_vals.push(',');
				alias_vals
					.push_str(&format!(r#"alias = "{}""#, heck::ToLowerCamelCase::to_lower_camel_case(name.as_str())));

				let alias = TokenStream2::from_str(alias_vals.as_str()).unwrap();
				let serde_alias: Attribute = syn::parse_quote! {
					#[serde(#alias)]
				};
				let unquoted_name = TokenStream2::from_str(name.as_str()).unwrap();

				quote! {
					#serde_alias
					#unquoted_name: #ty,
				}
			});
```

Reply:
I adopted the `RpcFnArg::name()` API.

I tested this change, it does not work because `RpcFnArg::name()` can be _any_ string, such as `"type"`, `"in"`, `"for"` etc.
If it was `"type"` the struct would be invalid because `"type"` is a reserved keyword.
Leveraging serde here gives us all possible strings that can be used with serde.